src/bootstrap.ts

```diff
@@ -54,70 +64,53 @@ async function run(
   }
 
   log.info('-----');
-  log.info(`Total packages   ${totalDocs}`);
+  log.info(chalk.yellowBright`Total packages: ${totalDocs}`);
   log.info('-----');
 
-  let lastProcessedId = state.bootstrapLastId;
-  do {
-    loopStart = Date.now();
-
-    lastProcessedId = await loop(lastProcessedId, stateManager, bootstrapIndex);
-  } while (lastProcessedId !== null);
-
-  log.info('-----');
-  log.info('⛷   Bootstrap: done');
-  await stateManager.save({
-    bootstrapDone: true,
-    bootstrapLastDone: Date.now(),
+  prefetcher = new Prefetcher({
```

Comment:
this utility sounds like it could be something built into async already, do you know if it exists?

Reply:
independently they are some functions that could work like `whilst` + `cargo` or `queue` + `until` 
I don't think there is one function that execute until a max and continue when an other condition occurs.

If I use `until` I still the code around that limits the prefetch, wait between iteration, save the offset and the fetch itself. 
I might save 10lines top